### ISSUE TITLE
cannot control how thumbnail is derived from image

### ISSUE THREAD

**Issue by OrenBochman** [jsilvermist/sl-gallery#2]
images that are portrait orientated are automatically cropped. The resulting thumbnail might be missing the top or bottom. The cropped thumbnail is sometimes not representative of the original.

**Comment by jsilvermist** [comment]
This was a choice made to allow good looking photo grids with minimal effort, and to prevent stretching of images.

To change the aspect to, for example, show all images in portrait, you can use this formula to set the aspect ratio, or simply specify a percentage:
```css
sl-gallery {
  /* The following will change the aspect ratio to use a 9 by 16 aspect ratio */
  --sl-gallery-item-height: calc(1 / (9 / 16) * 100%);
}
```

If you want to change this to show the entire image without stretching, you can use a simple bit of css to change the way the photos are displayed:
```css
sl-gallery-image {
  background-size: contain;
  background-repeat: no-repeat;
}
```
Or to show the entire image by stretching it to match the grid:
```css

sl-gallery-image {
  background-size: 100% 100%;
}
```

Note: While you are currently required to add `no-repeat`, this was an oversight and will be added by default in a future release.

Also, a simple way to select only a single gallery is to use the prefix as a selector, for example:
```css
sl-gallery[prefix="some-portrait-gallery-prefix"] {
  --sl-gallery-item-height: calc(1 / (9 / 16) * 100%);
}
```
Of course, you can also simply use `id`'s like you normally would to select something specific.

**Comment by jsilvermist** [comment]
Published 1.0.2 with `background-repeat: no-repeat;` in preparation for Polymer 3 support.
